provider/maas/devices.go

```diff
@@ -351,6 +351,7 @@ func (env *maasEnviron) createAndPopulateDevice(params deviceCreatorParams) (gom
 		for i, iface := range interface_set {
 			names[i] = iface.Name()
 		}
+		device.Delete()
```

Comment:
It'd be better to do this (and the others) in a deferred function, otherwise there's a risk that someone else adds another error path but forgets the `device.Delete()`.

Something like this after the `CreateDevice` call
``` go
defer func() {
    if err != nil {
        device.Delete()
    }
}()
```

(You'll need to name the error return.)